### PR TITLE
Signup page

### PR DESCRIPTION
**Pull request by muratcan23**

### CODE REVIEW

File: src/pages/Register/Components/BgPic.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      templateRows="repeat(2, 1fr)"
+      templateColumns="repeat(2, 1fr)"
+      maxWidth="400"
+      maxH="700"
+    >
+      <Box padding="1">
+        <Image src="https://d3ptyyxy2at9ui.cloudfront.net/assets/images/ae01c415678c90d0c606bc6917758502.jpg" />
+        <Box mt="5" ml="10">
+          <Text as="b" fontSize="sm">
+            30 million+{" "}
```
Comment (kod-man) [inline src/pages/Register/Components/BgPic.tsx:16]: remove unused symbols {}

File: src/pages/Register/Components/BgPic.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          </Text>
+          <Text fontSize="sm">app dowloads </Text>
+        </Box>
+      </Box>
+
+      <Box padding="1">
+        <Image src="https://d3ptyyxy2at9ui.cloudfront.net/assets/images/573fa908fbccdf1600f90f16a0c2e311.jpg" />
+        <Box margin="5" ml="10">
+          <Text as="b" fontSize="sm">
+            {" "}
```
Comment (kod-man) [inline src/pages/Register/Components/BgPic.tsx:26]: same

File: src/pages/Register/Components/BgPic.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          </Text>
+          <Text fontSize="sm">tasks completed</Text>
+        </Box>
+      </Box>
+
+      <Box padding="1">
+        <Image src="https://d3ptyyxy2at9ui.cloudfront.net/assets/images/ba90575091d9eafd60f851f7297cb202.jpg" />
+        <Box margin="5" ml="10">
+          <Text as="b" fontSize="sm">
+            1 million+{" "}
```
Comment (kod-man) [inline src/pages/Register/Components/BgPic.tsx:47]: same

File: src/pages/Register/Components/BgPic.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+          </Text>
+          <Text fontSize="sm">pro users </Text>
+        </Box>
+      </Box>
+    </Grid>
+  );
+};
+
+export default BgPics;
```
Comment (kod-man) [inline src/pages/Register/Components/BgPic.tsx:56]: As you can see in this file, you are repeating yourself.

   ```
<Box padding="1">
        <Image src="https://d3ptyyxy2at9ui.cloudfront.net/assets/images/ae01c415678c90d0c606bc6917758502.jpg" />
        <Box mt="5" ml="10">
          <Text as="b" fontSize="sm">
            30 million+{" "}
          </Text>
          <Text fontSize="sm">app dowloads </Text>
        </Box>
      </Box>
```

Create a generic component and pass the info as a prop.

```
   <Box padding="1">
        <Image src={props} />
        <Box mt="5" ml="10">
          <Text as="b" fontSize="sm">
       {props}
          </Text>
          <Text fontSize="sm">{props}</Text>
        </Box>
      </Box>
```

File: src/pages/Register/Components/ContinueWiths.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        <FaApple />
+        <Text as="b" ml="2" color="black">
+          Continue with Apple
+        </Text>
+      </Button>
+    </VStack>
+  );
+};
+
+export default ContinueWiths;
```
Comment (kod-man) [inline src/pages/Register/Components/ContinueWiths.tsx:34]: This part also can be a generic component for all icons. Their style is almost identical.
```

   <Button width="350px" colorScheme="gray" size="md" variant="outline">
        <FaApple />
        <Text as="b" ml="2" color="black">
          Continue with Apple
        </Text>
      </Button>
```

File: src/pages/Register/Components/Todoist.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { HStack, Box, Text } from "@chakra-ui/react";
+import React from "react";
```
Comment (kod-man) [inline src/pages/Register/Components/Todoist.tsx:2]: remove unused imports

File: src/pages/Register/Components/ContinueWiths.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import { VStack, Button, Box, Text } from "@chakra-ui/react";
```
Comment (kod-man) [inline src/pages/Register/Components/ContinueWiths.tsx:1]: unused imports

**Review by kod-man — COMMENTED** [review]
There is a settings in Vscode that removes unused imports when you hit the save. Investigate it :)

File: src/pages/Register/Components/BgPic.tsx (hunk 3 above)
Comment (muratcan23) [reply src/pages/Register/Components/BgPic.tsx:47]: ereased   {" "} symbols

File: src/pages/Register/Components/BgPic.tsx (hunk 2 above)
Comment (muratcan23) [reply src/pages/Register/Components/BgPic.tsx:26]: ereased   {" "} symbols

File: src/pages/Register/Components/BgPic.tsx (hunk 1 above)
Comment (muratcan23) [reply src/pages/Register/Components/BgPic.tsx:16]: ereased   {" "} symbols

File: src/pages/Register/index.tsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+            <Text as="b" fontSize="3xl">
+              Sign up
+            </Text>
+            <ContinueWiths />
+            <Divider orientation="horizontal" />
+
+            <Stack width="100%" spacing={3}>
+              <Input
+                size="lg"
+                pr="4.5rem"
```
[the remark this answers is not in the document]
Comment (muratcan23) [reply src/pages/Register/index.tsx:38]: changed from   `pr="4.5rem"`  => to ` pr="4"`

File: src/pages/Register/index.tsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+} from "@chakra-ui/react";
+import React from "react";
+import BgPics from "./Components/BgPic";
+import ContinueWiths from "./Components/ContinueWiths";
+import Todoist from "./Components/Todoist";
+import { AiOutlineEyeInvisible } from "react-icons/ai";
+
+function RegisterPage() {
+  return (
+    <Flex mr="40" ml="40" flexDirection="column">
```
[the remark this answers is not in the document]
Comment (muratcan23) [reply src/pages/Register/index.tsx:24]: swapped with mx

File: src/pages/Register/index.tsx [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+  HStack,
+  Link,
+  VStack,
+  Text,
+  InputGroup,
+  Input,
+  InputRightElement,
+  Button,
+  Flex,
+  Spacer,
```
[the remark this answers is not in the document]
Comment (muratcan23) [reply src/pages/Register/index.tsx:13]: spacer and Hstack romved

File: src/pages/Register/Components/ContinueWiths.tsx (hunk 7 above)
Comment (muratcan23) [reply src/pages/Register/Components/ContinueWiths.tsx:1]: unused box deleted

File: src/pages/Register/Components/Todoist.tsx (hunk 6 above)
Comment (muratcan23) [reply src/pages/Register/Components/Todoist.tsx:2]: unused box rmoved

File: src/pages/Register/Components/BgPic.tsx (hunk 4 above)
Comment (muratcan23) [reply src/pages/Register/Components/BgPic.tsx:55]: i didnt understand what am i supposed to do or not to do?

Comment (muratcan23) [reply src/pages/Register/Components/BgPic.tsx:44]: i think i got it . with ur props video's help i did used props and no more repeats.

**Review by kod-man — APPROVED** [review]

**Pull request merged**